lib/Error.js

```diff
@@ -1,94 +1,174 @@
 'use strict';
 
-const utils = require('./utils');
-
-module.exports = _Error;
-
 /**
- * Generic Error klass to wrap any errors returned by stripe-node
+ * Generic Error class to wrap any errors returned by stripe-node
  */
-function _Error(raw) {
-  this.populate(...arguments);
-  this.stack = new Error(this.message).stack;
-}
+class GenericError extends Error {
+  constructor(raw) {
+    super(arguments[1] || raw.message);
```

Comment:
hmm, I think this isn't quite right... 
Would have unexpected behavior in cases like these 
```
new StripeError(false, 'a string')
new StripeError('a string')
new StripeError({some: 'object'}, {another: 'object'})
```
eg; I think this might set type or even throw in places that the old way would not have...
getting super edge-casey here of course. 

I think this could maybe be solved by overriding the constructor to be more sane in StripeError, or with another layer of indirection...

Reply:
That's really super edge cases, particularly because original implementation doesn't do any typecheck and we can run in the same problems. In Javascript you always can construct an object with some property getter/setter or well known symbol implementation that will throw any function.

But ok. What is the _expected behavior_ in your examples? To throw or to ignore invalid parameters?

Reply:
Assuming expected behavior is like a standard errors (never throw, use as much information from params as possible) I've added typechecks and edge cases tests.
It doesn't make code any more beautiful, but bulletproof!